src/region.js

```diff
@@ -164,11 +164,17 @@ const Region = MarionetteObject.extend({
 
   // Restore the region's element in the DOM.
   _restoreEl() {
-    if (!this.currentView) {
+    // There is nothing to replace
+    if (!this._isReplaced) {
```

Comment:
OK. Last attempt at "fixing" an issue that I saw.
There are places where this logic happened but I didn't see it making sense to only apply it there. I figure it should always apply in here to prevent unnecessary replacement. Naturally, this assumes `_isReplaced` was not modified externally.